middleware/logger.go

```diff
@@ -43,7 +43,7 @@ func RequestLogger(f LogFormatter) func(next http.Handler) http.Handler {
 				entry.Write(ww.Status(), ww.BytesWritten(), ww.Header(), time.Since(t1), nil)
 			}()
 
-			next.ServeHTTP(ww, WithLogEntry(r, entry))
+			next.ServeHTTP(ww.Unwrap(), WithLogEntry(r, entry))
```

Comment:
Please, correct me if I'm wrong, but this will pass the original `w ResponseWriter` down to the Middleware chain and to the final Handler. It's like passing `w` variable directly.

Thus, the `ww.Status()` and `ww.BytesWritten()` above won't work anymore.

Reply:
yes, you are right, i missed that. to fix my issue with the hijacker interface i have another solution: 

i saw, that the `NewWrapResponseWriter` function can return a httpFancyWriter. the fancywriter is only chosen, if the original response writer has a flusher, hijacker and readerFrom.
in my specific case, a flushWriter will be returned, which does not support hijack.

there are multiple options here. 
* i could add the hijack method to the flushwriter.
* i could make a new writer that supports hijack and flush
* i could remove the flushwriter and use the httpFancyWriter if one of the interfaces (flusher, hijacker, readerFrom) is supported. in wrap_writer.go line 28. `if fl && hj && rf `-> `if fl || hj || rf`

i would go for the third option, but i do not know if there is a specific intention for the flushWriter.

Reply:
3. sounds good to me too